td.vue/src/views/ImportModel.vue

```diff
@@ -61,6 +61,8 @@ import { getProviderType } from '@/service/provider/providers.js';
 import openThreatModel from '@/service/openThreatModel.js';
 import TdFormButton from '@/components/FormButton.vue';
 import tmActions from '@/store/actions/threatmodel.js';
+import { isValidSchema } from '../service/ajv-SchemaVerification';
```

Comment:
It would be good to tidy up the comments so here is a suggestion:

```
% git diff td.vue/src/views/ImportModel.vue
--- a/td.vue/src/views/ImportModel.vue
+++ b/td.vue/src/views/ImportModel.vue
@@ -136,7 +136,7 @@ export default {
         },
         onImportClick(fileName) {
             let jsonModel;
-            // check for JSON syntax errors, schema errors come later
+            // check for JSON syntax errors, then validate with schema
             try {
                 jsonModel = JSON.parse(this.tmJson);
             } catch (e) {
@@ -145,15 +145,11 @@ export default {
                 return;
             }
 
-            // ToDo: need to catch invalid threat model schemas, possibly using npmjs.com/package/ajv
-
             console.log(jsonModel)
             if(!isValidSchema(jsonModel)){
-                return 
+                return;
             }
-            
-
             // Identify if threat model is in OTM format and if so, convert OTM back to dragon format
```

Reply:
I have made the necessary code refactorization. I had already made these changes, but I forgot to push the refactored code. Sorry for the inconvenience.

Reply:
> I have made the necessary code refactorization. I had already made these changes, but apparently forgot to push the refactored code. Sorry for the inconvenience!

Reply:
I have made the necessary code refactorization. I had already made these changes, but apparently forgot to push the refactored code. Sorry for the inconvenience!